pandas/tests/extension/base/reshaping.py

```diff
@@ -170,3 +170,11 @@ def test_merge(self, data, na_value):
                  [data[0], data[0], data[1], data[2], na_value],
                  dtype=data.dtype)})
         self.assert_frame_equal(res, exp[['ext', 'int1', 'key', 'int2']])
+
+    def test_merge_on_int_array(self, df_merge_on_int_array):
```

Comment:
this needs to be generic, see the test_merge example. it should work for any array type.